frappe/utils/backups.py

```diff
@@ -104,14 +104,43 @@ def zip_files(self):
 
 			print('Backed up files', os.path.abspath(backup_path))
 
-	def take_dump(self):
+	def take_dump(self, encrypt=False, automated_backup=None):
 		import frappe.utils
 
 		# escape reserved characters
 		args = dict([item[0], frappe.utils.esc(item[1], '$ ')]
 			for item in self.__dict__.copy().items())
 
-		cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s > %(backup_path_db)s """ % args
+		cmd_string = ""
+		if not automated_backup:
+			if encrypt:
+				file_name = args['backup_path_db'].split('/')
+				file_name = file_name[4]
+				file_name = file_name.replace('.sql', '.xb.enc')
+				args['backup_path_db'] = args['backup_path_db'].replace(args['backup_path_db'].split('/')[4], file_name)
+				args['encryption_password'] = frappe.utils.password.get_decrypted_password("System Settings",
+																						   "System Settings",
+																						   "confirm_password_encryption",
+																						   False)
+				cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s --triggers --routines | openssl enc -aes-256-cbc -salt -k %(encryption_password)s > %(backup_path_db)s""" % args
+			else:
+				cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s > %(backup_path_db)s """ % args
+
+		elif hasattr(frappe.get_doc("System Settings"), "enable_backup_encryption"):
+			if frappe.get_doc("System Settings").enable_backup_encryption:
```

Comment:
When we added the needed fields into erpnext doctype like System Settings, we only used one current running site, so this site only has these fields for the app Erpnext that is currently installed on that currently running site. so when the auto backup is running through cron job it gets an error cause the other sites' Erpnext app don't have these fields. So in order to avoid the error during the auto backup for all of the site, we need to check first if the current site that will be dumped has  the field "enable_backup_encryption" and check if it is selected before we decide to encrypt the output file.

Reply:
Hi, any updates for this? thanks.

Reply:
Hey. Sorry for not being clear earlier.

Two things here
But running sites whose database is not up to date is an anti-pattern anyway.

So Please get rid of 
```python3
		elif hasattr(frappe.get_doc("System Settings"), "enable_backup_encryption"):
			if frappe.get_doc("System Settings").enable_backup_encryption:
```

Reply:
Hi, sorry. Actually, we need this line. It just has a missing condition that's why it looks like it's not necessary. The only thing that is not necessary for this line is this ```elif hasattr(frappe.get_doc("System Settings"), "enable_backup_encryption"):``` the remaining lines is useful for the backup that will be processed through a cronjob. so if it's through a cron job, we'll need to check if the option for enabling the encryption is selected. We just need to add the condition to check first if it's automated using automated_backup variable and then the rest of the lines of this condition will follow.